src/riak_kv_pb_bucket.erl

```diff
@@ -66,90 +66,87 @@ init() ->
-                error ->
-                    {error,
-                        {format, "No bucket-type named '~s'", [Type]}, State}
-            end;
-        _ ->
-            {error,
-                riak_core_util:job_class_disabled_message(binary, Class),
-                State}
-    end;
 
 %% this should remain for backwards compatibility
 process(rpblistbucketsreq, State) ->
     process(#rpblistbucketsreq{}, State);
 
-%% Start streaming in list keys
-process(#rpblistkeysreq{type = Type, bucket = B, timeout = T} = Req,
-        #state{client = Client} = State) ->
-    % at present list-keys always streams
-    Class = {riak_kv, stream_list_keys},
+%% @doc process/2 callback. Handles an incoming request message.
+process(Req, State) ->
+    {Class, Listing} = determine_class_and_listing(Req),
+    Accept = determine_accept_and_report_job_disposition(Class),
+    case {Accept, Listing} of
+        {true, buckets} ->
+            maybe_do_list_buckets(Req, State);
+        {true, keys} ->
+            %% at present list-keys always streams
+            %% Start streaming in list keys
+            maybe_stream_list_keys(Req, State);
+        {false, _} ->
+            error_accept(Class, State)
+    end.
+
+
+determine_accept_and_report_job_disposition(Class) ->
     Accept = riak_core_util:job_class_enabled(Class),
     _ = riak_core_util:report_job_request_disposition(
-            Accept, Class, ?MODULE, process, ?LINE, protobuf),
-    case Accept of
-        true ->
-            %% stream_list_keys results will be processed by process_stream/3
-            case check_bucket_type(Type) of
-                {ok, GoodType} ->
-                    Bucket = maybe_create_bucket_type(GoodType, B),
-                    {ok, ReqId} = Client:stream_list_keys(Bucket, T),
-                    {reply, {stream, ReqId},
-                        State#state{req = Req, req_ctx = ReqId}};
-                error ->
-                    {error,
-                        {format, "No bucket-type named '~s'", [Type]}, State}
-            end;
-        _ ->
-            {error,
-                riak_core_util:job_class_disabled_message(binary, Class),
-                State}
+        Accept, Class, ?MODULE, process, ?LINE, protobuf),
+    Accept.
+
+determine_class_and_listing(#rpblistbucketsreq{stream = true}) ->
+    {{riak_kv, stream_list_buckets}, buckets};
+determine_class_and_listing(#rpblistbucketsreq{stream = false}) ->
+    {{riak_kv, list_buckets}, buckets};
+determine_class_and_listing(#rpblistkeysreq{}) ->
+    {{riak_kv, stream_list_keys}, keys}.
+
+maybe_stream_list_keys(#rpblistkeysreq{type = Type, bucket = B, timeout = T} = Req,
+                       #state{client = Client} = State) ->
+    case check_bucket_type(Type) of
+        {ok, GoodType} ->
+            Bucket = maybe_create_bucket_type(GoodType, B),
+            {ok, ReqId} = Client:stream_list_keys(Bucket, T),
+            {reply, {stream, ReqId},
```

Comment:
Small formatting nitpick, but this line no longer really needs to be split up across multiple lines now that it's been cut and pasted out of the nested case statement where it used to be.

Same applies for the bodies of `error_no_bucket_type` and `error_accept` below, as well as the final clause of `validate_request` farther down.